api/client/crop_model.py

```diff
@@ -69,3 +70,51 @@ def compute_crop_weighted_series(self,
             # TODO: change metric to reflect it is weighted in this copy
             series_list.append(series)
         return pandas.concat(series_list)
+
+
+    def growing_degree_days(self, region_name, base_temperature,
+                            start_date, end_date):
+        """Get Growing Degree Days (GDD) for a region.
+
+        Growing degree days (GDD) are a weather-based indicator that
+        allows for assessing crop phenology and crop development,
+        based on heat accumulation. GDD for one day is defined as
+        [(T_max + T_min)/2 - T_base], and the GDD over a longer time
+        interval is the sum of the GDD over all days in the interval.
+
+        The region can be any region of the Gro regions, from a point
```

Comment:
What if only Tmean is available for a region (say using LST) while daily Tmin and Tmax were not available or incomplete (ghcn stations)?

Reply:
Yes good question.  The idea is that we use find_data_series to get the best version of GDD, and that depends on the combination of search ranking and series availability.

If the region is specific and the daily max and min exists for that exact region,  e.g. a weather station  with GHCN data, then it does exactly what you would expect.   If not , then it depends on what is avialable.. If we don't have max and min temp but we do have the daily mean temperature, then  that should be the closest matching item and metric, and taking the mean of that should be ok. Or if we have more samples uniformly throughout the day  e.g. temperature every hour, then takign the average ofa all of them should be ok too. But note that *land surface* temperature is different than *air*  temperature so that would be wrong. 
Similarly, if the region is higher level, then it should give preference to weather stations in that district, rather than picking other lst at the district level.

Updataed the code to make the generalization clearer.

Reply:
Often there are some GHCN stations with available data. The issue is that GHCN station data tend to be highly sparse temporally. For example, there might be data for 2 years or a few days a year, while the desired time period is spreading over 10 years. I think there needs to be a judgement call on a minimum threshold of temporal coverage for a region required to use the data from a source. Like you said, air temperature is very different from land surface temperature, so we need to be consistent with choosing one temperature source for a given spatial extent (likely a list of districts or provinces) and time period.

Reply:
Yup, I was planning on adding coverage threshold later. But I've added it to this PR.